### PR TITLE
docstrings for compiled builtin modules

### LINKED ISSUE #160: Docstrings for standard library
One feature that Pylance has that I'm sorely missing is that they have detailed docstrings for the Python standard library. Pyright seems to have very few, if any. I'm not sure if Microsoft custom wrote the docs that they use for Pylance, or where they come from. But if they're available somewhere, it would be great to have that as part of this LSP, for Pylance parity.

### PR DESCRIPTION
**Pull request by DetachHead**
fixes #160 

a few things we need to address before merging tho:

- [x] this currently only generates stubs for the python version & platform build script is running on. need to set up github actions to run them on every platform and python version, then combine them at the end
- [x] currently this PR drops support for python 3.8 because docify doesn't support it. we should probably wait until 3.8 goes eol before merging this, or do you think it's feasible to add 3.8 support @AThePeanut4?
  - edit: turned out this was as easy as downgrading libcst to 1.1.0
- [x] doesn't work on the playground because it uses the npm package, and docify is currently only run in the pypi package
- [x] doesn't work when using the version of basedpyright bundled with the vscode extension for the same reason. (though not an issue if you have the pypi package installed and are using the `"fromEnvironment"` `importStrategy`)
- [x] document it, and also mention the docify repo if users need to generate docs for their own stubs
- [x] add tests

### CODE REVIEW

File: .github/workflows/mypy_primer_pr.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install -U pip
-          pip install git+https://github.com/detachhead/mypy_primer.git
+          pip install git+https://github.com/detachhead/mypy_primer.git@basedpyright-fix
```
Comment (DetachHead) [inline .github/workflows/mypy_primer_pr.yaml:54]: merge this and revert this

**Comment by AThePeanut4** [comment]
@DetachHead python 3.8 has a few other minor issues besides just the libcst version, I've pushed a fix to support 3.8 fully. I think you missed `3.8` in `based_build/generateAllDocstubs.sh

I've also added a `run` function that accepts kwargs rather than a list of string arguments, for easier usage when running programmatically.

**Comment by DetachHead** [comment]
oh oops, nice catch. thanks!

fixed in #423

**Comment by jfcherng** [comment]
I think docstrings can be added to the whole stdlib. Current it seems to be only done with `builtin.pyi`?

**Comment by DetachHead** [comment]
yeah i did that on purpose. i only made it run on compiled builtins because i figured running it on the entire stdlib would unnecessarily make it slower to build and increase the risk of something going wrong. since pyright can already see docstrings from all the other stdlib modules anyway i didn't see the point.

feel free to raise an issue if you disagree. i'm open to changing this behavior in the future

**Comment by jfcherng** [comment]
Ah, got it. We only need it for compiled modules. NVM.

**Comment by jfcherng** [comment]
> Ah, got it. We only need it for compiled modules. NVM.

Hmm... doesn't seem to be this case.

```py
import re

if m := re.search('foo', 'bar'):
    aa = m.group()
    #      ^^^^^ no docstring for `group`
```

**Comment by DetachHead** [comment]
oh man that's annoying, i didn't know there were modules that are only partially compiled. also it looks like pylance has the same issue

i opened #428 to track this

**Comment by YodaEmbedding** [comment]
Hover doesn't seem to work for docstrings that are missing from `basedpyright/dist/typeshed-fallback/stdlib/**.pyi`.

Non-working:
- [`re`](https://github.com/python/cpython/blob/v3.13.0/Lib/re/__init__.py#L17-L123)
- [`re.search`](https://github.com/python/cpython/blob/v3.13.0/Lib/re/__init__.py#L174-L177)
- [`asyncio.gather`](https://github.com/python/cpython/blob/v3.13.0/Lib/asyncio/tasks.py#L784-L813)

Working:
- builtins
- `re.Match.group`

**Pull request merged**